TeamSnapSDK/SDK/DataLayer/TSDKObjectsRequest.m

```diff
@@ -315,11 +319,15 @@ + (TSDKCollectionObject *)teamSnapObjectFromCollectionJSON:(TSDKCollectionJSON *
     static NSMutableDictionary *logHeadersForTypes;
     
     if (!unknownTypes) {
-        unknownTypes = [[NSMutableDictionary alloc] init];
+        dispatch_barrier_sync([self processingQueue], ^{
```

Comment:
This can be wrapped in ifdef debug..or event the DEBUGCURL flag. No reason to be in production..